runtime/src/main/java/org/corfudb/runtime/collections/PersistedStreamingMap.java

```diff
@@ -275,6 +278,27 @@ public void clear() {
         return resStream;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<V> scanAndFilter(Predicate<? super V> valuePredicate) {
+        return entryStream()
+                .map(Entry::getValue).filter(valuePredicate)
+                .collect(Collectors.toCollection(ArrayList::new));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Collection<Entry<K, V>> scanAndFilterByEntry(
+            Predicate<? super Entry<K, V>> entryPredicate) {
+        return entryStream()
+                .filter(entryPredicate)
+                .collect(Collectors.toCollection(ArrayList::new));
+    }
+
```

Comment:
Maybe we can do this in another patch, but we need to get rid of this.

If the table set cannot fit in memory, then its possible that the resulting set won't fit in memory.
Applications can just directly use the Stream exposed by entryStream.